staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -191,7 +191,15 @@ func (a *Authenticator) UpdateTransportConfig(c *transport.Config) error {
 		dial = (&net.Dialer{Timeout: 30 * time.Second, KeepAlive: 30 * time.Second}).DialContext
 	}
 	d := connrotation.NewDialer(dial)
+
+	a.mu.Lock()
+	defer a.mu.Unlock()
+	// Before replace with new connrotation dialer, close all existing connections.
+	if a.onRotate != nil {
+		a.onRotate()
```

Comment:
the documentation for CloseAll indicates new connections can be opened before CloseAll returns. doesn't that mean we should call the old onRotate method *after* assigning onRotate/Dial?